includes/REST_API/Publisher_Logos_Controller.php

```diff
@@ -0,0 +1,435 @@
+	 * Updates a publisher logo in the collection.
+	 *
+	 * Can only be used to make it the "active" one.
+	 *
+	 * @since 1.12.0
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return WP_REST_Response|WP_Error Response object on success, or WP_Error object on failure.
+	 */
+	public function update_item( $request ) {
+		$post = $this->get_publisher_logo( $request['id'] );
+
+		if ( is_wp_error( $post ) ) {
+			return $post;
+		}
+
+		if ( $request['active'] ) {
+			update_option( Settings::SETTING_NAME_ACTIVE_PUBLISHER_LOGO, $post->ID );
+		}
+
+		return $this->prepare_item_for_response( $post, $request );
+	}
+
+	/**
+	 * Get an existing publisher logo's post object, if valid.
+	 *
+	 * @since 1.12.0
+	 *
+	 * @param int $id Supplied ID.
+	 * @return WP_Post|WP_Error Post object if ID is valid, WP_Error otherwise.
+	 */
+	protected function get_publisher_logo( $id ) {
+		$publisher_logos = $this->filter_publisher_logos( (array) get_option( Settings::SETTING_NAME_PUBLISHER_LOGOS, [] ) );
+
+		$post = get_post( $id );
+
+		if ( ! $post || ! in_array( $post->ID, $publisher_logos, true ) ) {
+			return new WP_Error(
+				'rest_invalid_id',
+				__( 'Invalid ID', 'web-stories' ),
+				[ 'status' => 400 ]
+			);
+		}
+
+		return $post;
+	}
+
+	/**
+	 * Prepares a single publisher logo output for response.
+	 *
+	 * @since 1.12.0
+	 *
+	 * @param WP_Post         $post    Post object.
+	 * @param WP_REST_Request $request Request object.
+	 * @return WP_REST_Response Response object.
+	 */
+	public function prepare_item_for_response( $post, $request ): WP_REST_Response {
+		$fields = $this->get_fields_for_response( $request );
+
+		// Base fields for every post.
+		$data = [];
+
+		if ( rest_is_field_included( 'id', $fields ) ) {
+			$data['id'] = $post->ID;
+		}
+
+		if ( rest_is_field_included( 'title', $fields ) ) {
+			$data['title'] = get_the_title( $post->ID );
+		}
+
+		if ( rest_is_field_included( 'active', $fields ) ) {
+			$active_publisher_logo_id = absint( get_option( Settings::SETTING_NAME_ACTIVE_PUBLISHER_LOGO ) );
+			$data['active']           = $post->ID === $active_publisher_logo_id;
+		}
+
+		if ( in_array( 'url', $fields, true ) ) {
+			$data['url'] = wp_get_attachment_url( $post->ID );
+		}
+
+		/**
+		 * Wrapped response object.
+		 *
+		 * @var WP_REST_Response $response Response object.
+		 */
+		$response = rest_ensure_response( $data );
+
+		$links = $this->prepare_links( $post );
+		$response->add_links( $links );
+
+		return $response;
+	}
+
+	/**
+	 * Filters publisher logos to remove non-existent or invalid ones.
+	 *
+	 * @param int[] $publisher_logos List of publisher logos.
+	 * @return int[] Filtered list of publisher logos.
+	 */
+	protected function filter_publisher_logos( $publisher_logos ): array {
+		return array_filter(
+			$publisher_logos,
+			static function( $logo ) {
+				$post = get_post( $logo );
+
+				return ! ( ! $post || 'attachment' !== $post->post_type );
+			}
+		);
```

Comment:
```suggestion
		return array_filter( $publisher_logos, 'wp_attachment_is_image' );
```

Would this be a little cleaner?

Reply:
It would!